catalyst-gateway/bin/src/service/common/types/cardano/asset_value.rs

```diff
@@ -41,6 +41,19 @@ static SCHEMA: LazyLock<MetaSchema> = LazyLock::new(|| {
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub(crate) struct AssetValue(i128);
 
+impl Display for AssetValue {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl AssetValue {
+    /// Performs safe addition that returns None instead of wrapping around on overflow.
+    pub(crate) fn checked_add(&self, v: &Self) -> Option<Self> {
```

Comment:
Same with asset value.  It's not practical or feasible to overflow a u64 of any asset.  There are not enough.  So saturate, log an error in the saturation add function if it does saturate which it never will, and Dave the caller the headache of handling an error that's actually impossible.